diff --git a/node_modules/helios-core/dist/dl/DownloadEngine.js b/node_modules/helios-core/dist/dl/DownloadEngine.js
index 922cd1a..14edf00 100644
--- a/node_modules/helios-core/dist/dl/DownloadEngine.js
+++ b/node_modules/helios-core/dist/dl/DownloadEngine.js
@@ -59,7 +59,7 @@ async function downloadQueue(assets, onProgress) {
         };
     };
     const wrap = (asset) => downloadFile(asset.url, asset.path, onEachProgress(asset));
-    const q = fastq.promise(wrap, 15);
+    const q = fastq.promise(wrap, 30);
     const promises = assets.map(asset => q.push(asset)).reduce((acc, p) => ([...acc, p]), []);
     await Promise.all(promises);
     return receivedTotals;
diff --git a/node_modules/helios-core/dist/dl/distribution/DistributionIndexProcessor.js b/node_modules/helios-core/dist/dl/distribution/DistributionIndexProcessor.js
index f09b6b4..74427c6 100644
--- a/node_modules/helios-core/dist/dl/distribution/DistributionIndexProcessor.js
+++ b/node_modules/helios-core/dist/dl/distribution/DistributionIndexProcessor.js
@@ -45,7 +45,7 @@ class DistributionIndexProcessor extends IndexProcessor_1.IndexProcessor {
         await this.loadModLoaderVersionJson();
     }
     async validateModules(modules, accumulator) {
-        for (const module of modules) {
+        await Promise.all(modules.map(async (module) => {
             const hash = module.rawModule.artifact.MD5;
             if (!await (0, FileUtils_1.validateLocalFile)(module.getPath(), Asset_1.HashAlgo.MD5, hash)) {
                 accumulator.push({
@@ -60,7 +60,7 @@ class DistributionIndexProcessor extends IndexProcessor_1.IndexProcessor {
             if (module.hasSubModules()) {
                 await this.validateModules(module.subModules, accumulator);
             }
-        }
+        }));
     }
     async loadModLoaderVersionJson() {
         const server = this.distribution.getServerById(this.serverId);
diff --git a/node_modules/helios-core/dist/dl/mojang/MojangIndexProcessor.js b/node_modules/helios-core/dist/dl/mojang/MojangIndexProcessor.js
index 37d1008..fe300e1 100644
--- a/node_modules/helios-core/dist/dl/mojang/MojangIndexProcessor.js
+++ b/node_modules/helios-core/dist/dl/mojang/MojangIndexProcessor.js
@@ -165,7 +165,8 @@ class MojangIndexProcessor extends IndexProcessor_1.IndexProcessor {
     async validateAssets(assetIndex) {
         const objectDir = (0, path_1.join)(this.assetPath, 'objects');
         const notValid = [];
-        for (const assetEntry of Object.entries(assetIndex.objects)) {
+        const entries = Object.entries(assetIndex.objects);
+        await Promise.all(entries.map(async (assetEntry) => {
             const hash = assetEntry[1].hash;
             const path = (0, path_1.join)(objectDir, hash.substring(0, 2), hash);
             const url = `${MojangIndexProcessor.ASSET_RESOURCE_ENDPOINT}/${hash.substring(0, 2)}/${hash}`;
@@ -179,13 +180,13 @@ class MojangIndexProcessor extends IndexProcessor_1.IndexProcessor {
                     path
                 });
             }
-        }
+        }));
         return notValid;
     }
     async validateLibraries(versionJson) {
         const libDir = (0, FileUtils_1.getLibraryDir)(this.commonDir);
         const notValid = [];
-        for (const libEntry of versionJson.libraries) {
+        await Promise.all(versionJson.libraries.map(async (libEntry) => {
             if ((0, MojangUtils_1.isLibraryCompatible)(libEntry.rules, libEntry.natives)) {
                 let artifact;
                 if (libEntry.natives == null) {
@@ -212,7 +213,7 @@ class MojangIndexProcessor extends IndexProcessor_1.IndexProcessor {
                     });
                 }
             }
-        }
+        }));
         return notValid;
     }
     async validateClient(versionJson) {
diff --git a/node_modules/helios-core/dist/dl/receivers/FullRepairReceiver.js b/node_modules/helios-core/dist/dl/receivers/FullRepairReceiver.js
index 89fe157..2fff172 100644
--- a/node_modules/helios-core/dist/dl/receivers/FullRepairReceiver.js
+++ b/node_modules/helios-core/dist/dl/receivers/FullRepairReceiver.js
@@ -31,7 +31,13 @@ class FullRepairReceiver {
             }
             if (error instanceof got_1.HTTPError) {
                 log.debug('Response Details:');
-                log.debug('Body:', error.response.body);
+                if (error.response.body != null) {
+                    let body = `${error.response.body}`;
+                    if (body.length > 500) {
+                        body = body.substring(0, 500) + '... (truncated)';
+                    }
+                    log.debug('Body:', body);
+                }
                 log.debug('Headers:', error.response.headers);
                 return `Error during request (HTTP Response ${error.response.statusCode})`;
             }
